server/server.go

```diff
@@ -405,6 +444,16 @@ func (s *Server) GracefulDown() {
 	}
 }
 
+// CleanupSocketFile cleans up socket file if it was created.
+func (s *Server) CleanupSocketFile() {
+	if s.cfg.Socket != "" {
+		log.Infof("[server] removing socket file [%s]", s.cfg.Socket)
+		if err := os.Remove(s.cfg.Socket); err != nil {
+			log.Warningf("[server] failed to remove socket file! err: %s", err)
```

Comment:
I think this should be logged at the Error level since there is clearly an error occurring.